### PR TITLE
Add Import-LocalizedData implicit Localization fallback

### LINKED ISSUE #19923: Import-LocalizedData implicit Localization fallback
### Summary of the new feature / enhancement

`Import-LocalizedData` should implement a default culture fallback when `-UICulture` is not specified to simulate the functionality of the command in Windows PowerShell.

PR: #19896

### Proposed technical implementation details (optional)

`Import-LocalizedData` is used by several of the modules released by other teams in Microsoft, most of which as cdxml with a small amount of other code.

For example, in PowerShell 7, attempting to import the SmbShare module will result in an error:

```powershell
PS> Import-Module SmbShare
Import-LocalizedData: Cannot find the PowerShell data file 'SmbLocalization.psd1' in directory
'C:\Windows\system32\WindowsPowerShell\v1.0\Modules\SmbShare\en-GB', or in any parent culture directories.
```

Or more directly:

```powershell
Push-Location C:\Windows\system32\WindowsPowerShell\v1.0\Modules\smbshare
Import-LocalizedData -BindingVariable _system_translations -fileName SmbLocalization.psd1 -ea stop
$_system_translations
Pop-Location
```

This error is not present in Windows PowerShell.

This error is caused by a difference in how .NET Core+ and .NET Framework handle the `CurrentUICulture` static property
In .NET Core and higher, the values in the parent list are arguably correct:

```powershell
$culture = [System.Globalization.CultureInfo]::CurrentUICulture
do { $culture.Name; $culture = $culture.Parent } while ($culture.Name)
```

For my culture of en-GB results in the following list:

```
en-GB
en
# <Invariant>
```

However, if the same test is repeated in Windows PowerShell `en-US` is implicitly added to the bottom of the list. This only happens when the culture is found using the static property `CultureInfo.CurrentUICulture` .

The result in Windows PowerShell is shown below.

```powershell
PS> $culture = [System.Globalization.CultureInfo]::CurrentUICulture
PS> do { $culture.Name; $culture = $culture.Parent } while ($culture.Name)
en-GB
en
en-US
en
# <Invariant>
```

A spot check of a few other non-en-US UI cultures shows similar results, but I haven't looked hard enough to find a simple way to change the current UI culture (in Windows PowerShell) to really prove that.

For `Import-LocalizedData` this represents an implicit fallback culture in Windows PowerShell. While this is arguably caused by a bug in .NET, the result is that modules were released with content that worked based on the presence of this bug.

`Import-LocalizedData` is used by a fairly broad set of modules in the `system32` directory, including:

```
Appx
AssignedAccess
BitLocker
DhcpServer
FailoverClusters
HgsClient
IpamServer
IscsiTarget
Microsoft.PowerShell.Archive
Microsoft.PowerShell.ODataUtils
MsDtc
NetworkSwitchManager
RemoteDesktop
ServerManager
SmbShare
StorageQoS
StorageReplica
```

I propose that when the `-UICulture` parameter is not supplied, that `en-US` should be added to the bottom of the list of parent cultures to search. This restores the behaviour of the command as it was in 5.1.

### PR DESCRIPTION
**Pull request by chrisdent-de**
# PR Summary

Fixes #19923

 - Adds a fallback on `en-US` UI culture to `Import-LocalizedData` when the `-UICulture` parameter is not specified.
 - Refactors I18n tests and adds new tests for behaviour change.

## PR Context

Certain shipped modules incorrectly handle localization which causes errors when importing into PowerShell. For example:

```powershell
PS> Import-Module SmbShare
Import-LocalizedData: Cannot find the PowerShell data file 'SmbLocalization.psd1' in directory
'C:\Windows\system32\WindowsPowerShell\v1.0\Modules\SmbShare\en-GB', or in any parent culture directories.
```

This problem is not present in Windows PowerShell. This failure is caused by a change in how .NET exposes current UI culture. What is likely a bug in .NET Framework has been fixed in .NET Core (and greater). In .NET Framework

In .NET Framework and Windows PowerShell the UI culture hierarchy derived from CurrentUICulture includes `en-US`:

```powershell
PS> $culture = [System.Globalization.CultureInfo]::CurrentUICulture
PS> do { $culture.Name; $culture = $culture.Parent } while ($culture.Name)

en-GB
en
en-US
en
```

In PowerShell 6+ and .NET Core+ this is fixed and the parent of the first `en` is invariant.

So while this is arguably a bug in source modules, it was not so in Windows PowerShell. PowerShell could conceivably handle the problem by automatically falling back on `en-US`.

This PR introduces a fallback culture into `Import-LocalizedData` when the `-UICulture` parameter is not supplied.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [x] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [x] Issue filed: #19923 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Comment by chrisdent-de** [comment]
@microsoft-github-policy-service agree company="D. E. Shaw & Co., L.P."

**Comment by chrisdent-de** [comment]
Converting to draft while I check non-existent culture behaviour on parameter binding in Linux.

File: test/powershell/Language/Scripting/I18n/I18n.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        ) {
+            param ( $Script )
+
+            { Invoke-Expression $Script } | Should -Throw -ErrorId 'CmdletNotInAllowedListForDataSection,Microsoft.PowerShell.Commands.InvokeExpressionCommand'
+        }
+    }
+
+    Context 'Data files' {
+        It 'Imports a data file' {
+            $daa
```
Comment (JamesWTruher) [inline test/powershell/Language/Scripting/I18n/I18n.Tests.ps1:53]: not sure what this is

**Review by JamesWTruher — COMMENTED** [review]
looks good, i think you left a test unfinished

**Comment by daxian-dbw** [comment]
@chrisdent-de Thanks for submitting this PR. However, it's required to open an issue first for discussing the solution before submitting the code change. Can you please open the corresponding issue?

**Comment by SteveL-MSFT** [comment]
@PowerShell/wg-powershell-cmdlets discussed this and agreed that when no `-UiCulture` is specified, it makes sense to fallback to `en-US`.

**Review by JamesWTruher — APPROVED** [review]
lgtm - thanks!

**Review by SeeminglyScience — APPROVED** [review]
LGTM! Thank you @chrisdent-de!

**Pull request merged**